blivet/partitioning.py

```diff
@@ -916,7 +918,16 @@ def allocate_partitions(storage, disks, partitions, freespace):
         if part_type == parted.PARTITION_EXTENDED and \
            part_type != _part.req_part_type:
             log.debug("creating extended partition")
-            add_partition(disklabel, free, part_type, None)
+            ext = add_partition(disklabel, free, part_type, None)
+
+            # extedned partition took all free space -- make the logical partition 1 MiB smaller
+            if aligned_size > (ext.geometry.length) * LINUX_SECTOR_SIZE - EXTENDED_METADATA_SIZE:
```

Comment:
I'm using the `align_size_for_disklabel` to align the new size (after the "extedned metada" subtraction). I thought it should be enough or not?